### DISCUSSION TITLE
Guardar estados y reiniciar el Buffer para volver a entrenar más tarde

### DISCUSSION THREAD

**Discussion by lucastack** [JavierBZ/Tarea3-IIC3697#11] [Q&A] [answered]
Hola!

En el lugar donde vivo el internet es muy inestable (no hay Wifi) y es común que después de un par de horas se caiga la conexión. Para resolver esto he tenido que ir guardando el progreso con torch.save y cargarlo después con torch.load. El problema que ocurre es que no puedo guardar el Buffer porque Pickle (que es lo que usa torch.save) se come la memoria RAM y termina la sesión de colab, perdiendo todo el progreso y obligando a tener que instanciar el Buffer vacío si quiero continuar el entrenamiento. 

Mi pregunta es: ¿es válido volver a llenar el Buffer como al principio y mantener todo lo demás? es decir, cargar la red y hacerla jugar por 50000 pasos (sin entrenar) al momento de retomar el entrenamiento en una época posterior.  

Saludos!

**Comment by JavierBZ** [comment, marked as answer]
Hola,

Este tema lo hablamos en ayudantía, si puedes volver a rellenar un nuevo buffer si reinicias tu entrenamiento. La idea es que lo llenes siguiendo la poílitica e-greedy donde lo dejaste y una vez completo entrenas nuevamente. Es posible que debido a esto tu curva de aprendizaje por épocas pierda continuidad, pero no hay problema con eso. Lo importante es que la red siga mejorando.

Espero te sirva!

**Reply by lucastack** [reply]
Si, estoy guardando la cantidad de pasos totales que llevo para mantener la política e-greedy donde la dejé. 

Muchas gracias Javier!

**Reply by tpcantergiani** [reply]
Has pensado en guardar solo una parte del buffer?
Por ejemplo los ultimos 50000, asi tienes las ultimas acciones elegidas, lo cual es un buen comienzo.
